### DISCUSSION TITLE
Ensure all nodes are connected

### DISCUSSION THREAD

**Discussion by Zonalds** [xyflow/xyflow#1943] [Q&A]
Is there a way to make sure that all nodes present are connected? I implemented a drag-and-drop for users to create their own flow but will like to ensure all nodes dragged into the flow are connected before submission.

Is this possible?

**Comment by moklick** [comment]
Hey @Zonalds 

you could check if all nodes have a connection: 

```js
const edges = elements.filter(isEdge);
const allNodesConnected = elements.filter(isNode).every(node => edges.find(edge => edge.source === node.id || edge.target === node.id));
```
